features/dd-sdk-android-trace/src/test/kotlin/com/datadog/opentelemetry/trace/OtelSpanTest.kt

```diff
@@ -0,0 +1,270 @@
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.verifyNoMoreInteractions
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class OtelSpanTest {
+
+    @Mock
+    lateinit var mockAgentTracer: AgentTracer.TracerAPI
+
+    @Mock
+    lateinit var mockAgentSpan: AgentSpan
+    lateinit var testedSpan: OtelSpan
+
+    @BeforeEach
+    fun `set up`() {
+        testedSpan = OtelSpan(mockAgentSpan, mockAgentTracer)
+    }
+
+    // region Init
+    @Test
+    fun `M set default attributes W init`() {
+        // Then
+        assertThat(testedSpan.isRecording).isTrue()
+        assertThat(testedSpan.statusCode).isEqualTo(StatusCode.UNSET)
+    }
+
+    @Test
+    fun `M return NOOP W invalid()`() {
+        // When
+        val result = OtelSpan.invalid()
+
+        // Then
+        assertThat(result).isSameAs(OtelSpan.NoopSpan.INSTANCE)
+    }
+
+    // endregion
+
+    @Test
+    fun `M add attribute W setAttribute`() {
+        // Given
+        val expectedTags = listOf(
+            "string" to "b",
+            "empty_string" to "",
+            "number" to 2L,
+            "boolean" to false,
+            "empty-string-attribute" to "",
+            "string-array.0" to "d",
+            "string-array.1" to "e",
+            "string-array.2" to "f",
+            "boolean-array.0" to false,
+            "boolean-array.1" to true,
+            "long-array.0" to 5L,
+            "long-array.1" to 6L,
+            "long-array.2" to 7L,
+            "long-array.3" to 8L,
+            "double-array.0" to 34.0,
+            "double-array.1" to 5.67,
+            "empty-array" to ""
+        )
+
+        // When
+        testedSpan.setAttribute("string", "b")
+        testedSpan.setAttribute("empty_string", "")
+        testedSpan.setAttribute("number", 2L)
+        testedSpan.setAttribute("boolean", false)
+        testedSpan.setAttribute(AttributeKey.stringKey("null-string-attribute"), null)
+        testedSpan.setAttribute(AttributeKey.stringKey("empty-string-attribute"), "")
+        testedSpan.setAttribute(AttributeKey.stringArrayKey("string-array"), listOf("d", "e", "f"))
+        testedSpan.setAttribute(AttributeKey.booleanArrayKey("boolean-array"), listOf(false, true))
+        testedSpan.setAttribute(AttributeKey.longArrayKey("long-array"), listOf(5L, 6L, 7L, 8L))
+        testedSpan.setAttribute(AttributeKey.doubleArrayKey("double-array"), listOf(34.0, 5.67))
+        testedSpan.setAttribute(AttributeKey.stringArrayKey("empty-array"), emptyList())
+        testedSpan.setAttribute(AttributeKey.stringArrayKey("null-array"), null)
+
+        // Then
+        expectedTags.forEach {
+            verify(mockAgentSpan).setTag(it.first, it.second)
+        }
+    }
+
+    // region setAttribute
+
+    // endregion
```

Comment:
empty region?

Reply:
nope it is just misplaced, I will correct it.